draft-oauth-ai-agents.md

```diff
@@ -0,0 +1,356 @@
+
+It is RECOMMENDED that Authorization Servers log all Agent-related actions and access events to support auditing and accountability.
+
+Authorization Servers MAY implement additional security controls such as rate limiting, IP allowlisting, or other measures to mitigate the risk of Agent misuse.
+
+## User Consent Experience
+
+Authorization Servers SHOULD clearly communicate to end-users the identity of the acting agent, the permissions requested, and the implications of consent (where applicable) during authorization prompts. Consent screens should be designed to be understandable, minimizing ambiguity about which agent is requesting access and what actions it may perform on the user’s behalf.
+
+# Resource Server Behavior
+
+Resource Servers MUST validate access tokens and their claims prior to granting access to protected resources. They SHOULD log all access attempts to support auditing and security.
+
+Specifically, Resource Servers MUST:
+
+* Validate that the access token is issued by a trusted Authorization Server and remains valid.
+* Verify that the token scopes authorize the requested resource access.
+
+If the access token is missing, invalid, or insufficient for the requested operation, the Resource Server MUST respond with an appropriate error, typically one of: HTTP 400 (Bad Request), HTTP 401 (Unauthorized), or HTTP 403 (Forbidden), and include a WWW-Authenticate header describing the error.
+
+It is RECOMMENDED that Resource Servers:
+
+* Log relevant token claims and request details to ensure traceability and auditability.
+* Use entity-describing claims (e.g., client_entity_type, sub_entity_type, client_parent, sub_parent) to make authorization decisions regarding Agent actions.
+
+Resource Servers MAY apply additional protections such as rate limiting or IP allowlisting to reduce abuse risks from agents.
+
+# Incremental Authorization
+
+Incremental authorization enables Agents to request only the permissions required for a specific task, deferring the acquisition of additional scopes until they are needed. This strategy reduces the risk of over-privileged access and aligns access token scopes with real-time operational context.
+
+Agents can dynamically determine the necessary permissions by relying on [Protected Resource Metadata] as defined in OAuth 2.0 Protected Resource Metadata {{RFC8414}} and responding to WWW-Authenticate challenges issued by Resource Servers. When a token is insufficient—due to missing scopes or delegation context—the Resource Server challenges the Agent (e.g., error="insufficient_scope"), prompting a subsequent authorization request to acquire the necessary access.
+
+Where supported, Authorization Servers MAY employ a CIBA-style (Client-Initiated Backchannel Authentication) authorization flow to facilitate end-user consent. This is particularly useful in non-interactive or delegated scenarios involving AI agents.
+
```

Comment:
+1 on using CIBA for out-of-band consent. 

There should be some security considerations on consent phishing attacks and mitigating them (perhaps citing the cross device security BCP - https://www.ietf.org/archive/id/draft-ietf-oauth-cross-device-security-09.html)